File: leetcode/429.n-ary-tree-level-order-traversal/ts/narylevelorder.ts

class _Node {
    val: number
    children: _Node[]

    constructor(v: number) {
        this.val = v;
        this.children = [];
    }
}

const levelOrder = (root: _Node | null): number[][] => {
    const res: number[][] = []
    const queue: _Node[] = []
    let front = 0

    if (root !== null) queue.push(root)

    while (queue.length > front) {
        const size = queue.length - front
        const layer: number[] = Array(size)

        for (let i = 0; i < size; i++) {
            const curr = queue[front++]

            layer[i] = curr.val

            for (const child of curr.children)
                if (child !== null) queue.push(child)
        }

        res.push(layer)
    }

    return res
}

export default levelOrder

File: leetcode/429.n-ary-tree-level-order-traversal/go/narylevelorder.go
package narylevelorder

import "container/list"

type Node struct {
	Val      int
	Children []*Node
}

func levelOrder(root *Node) [][]int {
	res := make([][]int, 0)
	queue := list.New()

	if root != nil {
		queue.PushBack(root)
	}

	for queue.Len() > 0 {
		size := queue.Len()

		layer := make([]int, size)
		for i := range size {
			curr := queue.Remove(queue.Front()).(*Node)

			layer[i] = curr.Val

			for _, n := range curr.Children {
				if n != nil {
					queue.PushBack(n)
				}
			}
		}

		res = append(res, layer)
	}

	return res
}
